src/pip/_internal/pyproject.py

```diff
@@ -174,4 +174,9 @@ def load_pyproject_toml(
         backend = "setuptools.build_meta:__legacy__"
         check = ["setuptools>=40.8.0", "wheel"]
 
+    if backend == "setuptools.build_meta:__legacy__" and not has_setup:
+        raise InstallationError(
+            f"File 'setup.py' not found for legacy project {req_name}."
```

Comment:
```suggestion
            f"File 'setup.py' not found for project {req_name} that uses the 'setuptools.build_meta:__legacy__' backend."
```

Reply:
IIRC setuptools wants to treat the legacy backend as an "internal" protocol between it and pip, so we probably shouldn't expose it in user-facing messages.